spec/ics-007-tendermint-client/README.md

```diff
@@ -0,0 +1,312 @@
+---
+ics: 7
+title: Tendermint Client
+stage: draft
+category: IBC/TAO
+kind: instantiation
+implements: 2
+author: Christopher Goes <cwgoes@tendermint.com>
+created: 2019-12-10
+modified: 2019-12-19
+---
+
+## Synopsis
+
+This specification document describes a client (verification algorithm) for a blockchain using Tendermint consensus.
+
+### Motivation
+
+State machines of various sorts replicated using the Tendermint consensus algorithm might like to interface with other replicated state machines or solo machines over IBC.
+
+### Definitions
+
+Functions & terms are as defined in [ICS 2](../ics-002-client-semantics).
+
+The Tendermint light client uses the generalised Merkle proof format as defined in [ICS 8](../ics-008-generalised-merkle-proof-format).
+
+### Desired Properties
+
+This specification must satisfy the client interface defined in ICS 2.
+
+## Technical Specification
+
+### Client state
+
+The Tendermint client state tracks the current validator set, latest height, and a possible frozen height.
+
+```typescript
+interface ClientState {
+  validatorSet: List<Pair<Address, uint64>>
+  latestHeight: uint64
+  frozenHeight: Maybe<uint64>
+}
+```
+
+### Consensus state
+
+The Tendermint client tracks the validator set hash & commitment root for all previously verified consensus states (these can be pruned after awhile).
+
+```typescript
+interface ConsensusState {
+  validatorSetHash: []byte
+  commitmentRoot: []byte
+}
+```
+
+### Headers
+
+The Tendermint client headers include a height, the commitment root, the complete validator set, and the signatures by the validators who committed the block.
+
+```typescript
+interface Header {
+  height: uint64
+  commitmentRoot: []byte
+  validatorSet: List<Pair<Address, uint64>>
+  signatures: []Signature
+}
+```
+
+### Evidence
+
+Tendermint client `Evidence` consists of two headers at the same height both of which the light client would have considered valid.
```

Comment:
Explain that the `Evidence` type is used for punishing client misbehaviour and freezing it if applicable.

Reply:
Added.